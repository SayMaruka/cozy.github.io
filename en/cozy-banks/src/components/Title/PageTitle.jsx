import React from 'react'
import PropTypes from 'prop-types'
import Title from 'components/Title/Title'
import { BarCenter } from 'components/Bar'
import useBreakpoints from 'cozy-ui/transpiled/react/hooks/useBreakpoints'
import styles from './PageTitle.styl'

export const BarTitle = ({ children }) => {
  return (
    <BarCenter>
      <Title className={styles.PageTitle}>{children}</Title>
    </BarCenter>
  )
}

const PageTitle = ({ children }) => {
  const { isMobile } = useBreakpoints()
  return isMobile ? <BarTitle>{children}</BarTitle> : <Title>{children}</Title>
}

PageTitle.propTypes = {
  children: PropTypes.node.isRequired
}

export default PageTitle
