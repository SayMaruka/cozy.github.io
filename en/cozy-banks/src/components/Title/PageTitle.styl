@require 'tools/mixins'
@require 'settings/breakpoints'

+small-screen()
    .PageTitle
        display     flex
        align-items center
        overflow hidden
