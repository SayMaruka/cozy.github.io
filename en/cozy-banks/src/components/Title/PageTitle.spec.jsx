import React from 'react'
import { shallow } from 'enzyme'
import PageTitle from './PageTitle'

describe('PageTitle', () => {
  it(`should display children`, () => {
    expect(
      shallow(<PageTitle>content</PageTitle>).getElement()
    ).toMatchSnapshot()
  })

  it(`should set color default`, () => {
    expect(
      shallow(<PageTitle color="default">content</PageTitle>).getElement()
    ).toMatchSnapshot()
  })

  it(`should set color primary`, () => {
    expect(
      shallow(<PageTitle color="primary">content</PageTitle>).getElement()
    ).toMatchSnapshot()
  })
})
