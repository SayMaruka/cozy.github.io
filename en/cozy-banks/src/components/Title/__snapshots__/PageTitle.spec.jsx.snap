// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle should display children 1`] = `
<Aware>
  <Title
    color="default"
  >
    content
  </Title>
</Aware>
`;

exports[`PageTitle should set color default 1`] = `
<Aware>
  <Title
    color="default"
  >
    content
  </Title>
</Aware>
`;

exports[`PageTitle should set color primary 1`] = `
<Aware>
  <Title
    color="primary"
  >
    content
  </Title>
</Aware>
`;
